src/hooks/loadables/useLoadSpendingLimits.ts

```diff
@@ -1,16 +1,21 @@
-import { useEffect } from 'react'
+import { useEffect, useMemo } from 'react'
 import useAsync, { type AsyncResult } from '../useAsync'
 import useSafeInfo from '../useSafeInfo'
 import { Errors, logError } from '@/services/exceptions'
 import type { SpendingLimitState } from '@/store/spendingLimitsSlice'
 import useChainId from '@/hooks/useChainId'
-import { useWeb3ReadOnly } from '@/hooks/wallets/web3'
+import { getWeb3, useWeb3ReadOnly } from '@/hooks/wallets/web3'
 import type { JsonRpcProvider } from '@ethersproject/providers'
 import { getSpendingLimitContract, getSpendingLimitModuleAddress } from '@/services/contracts/spendingLimitContracts'
 import type { AddressEx } from '@safe-global/safe-gateway-typescript-sdk'
 import { sameAddress } from '@/utils/addresses'
+import { ERC20__factory } from '@/types/contracts'
 import type { AllowanceModule } from '@/types/contracts'
+
 import { sameString } from '@safe-global/safe-core-sdk/dist/src/utils'
+import useBalances from '../useBalances'
+
+type TokenInfoFromBalances = { [address: string]: { symbol: string; decimals: number; logoUri?: string } }
```

Comment:
Let's reuse the type to be safe.

```suggestion
type TokenInfoFromBalances = { [address: string]: Omit<SpendingLimitState['token'], 'address'>
```